File: common_library.rb
#!/usr/bin/env ruby

################################################################################################################################################
#
# Common Ruby Library file to include some common functions
# Function list
# => ShellOut
#
################################################################################################################################################

require 'net/http'
require 'net/https'
require "uri"
require 'json'

def ShellOut(cmd, cwd=nil)
	@command = "#{ cmd }"
	if cwd !=nil and Dir.exist?(cwd)
		puts "use #{ cwd } as current dir"
		@command = "cd #{ cwd } && #{ cmd }"
	end
	@return_value = Hash.new
	puts "run command #{ @command }"
	@return_value["output"] = `#{ @command }`
	@return_value["success"] = $?.success?
	@return_value
end

def set_params
	@params = Hash.new
	ARGV.each do|arg|
		if !arg.include?("=")
			raise "Argument Exception: Unknown arguments #{ arg }"
		end
		param = arg.split("=",2)
		@params[param[0].downcase] = param[1]
	end
	@params
end

def http_post(post_url, data, headers)
	@uri = URI.parse(post_url)
	@https = Net::HTTP.new(@uri.host,@uri.port)
	@https.use_ssl = true
	@https.post(@uri.path,data.to_json,headers)
end

File: common_library.py
#! /usr/bin/env python

################################################################################################################################################
#
# Common python Library file to include some common functions
# Function list
# => ShellOut
#
################################################################################################################################################


import commands
import json
import os
import sys
import urllib
import urllib2

def ShellOut(cmd, cwd=None):
	command = cmd
	if cwd is not None and os.path.exists(cwd):
		print "use " + cwd + " as current dir"
		cmd = "cd " + cwd + " && " + cmd
	return_value = {}
	print "run command" + command
	status, output = commands.getstatusoutput(command)
	return_value["output"] = output
	return_value["status"] = status
	return return_value

def http_put(put_url, data, headers):
	request = urllib2.Request(put_url, data=data, headers = headers)
	request.get_method = lambda:'PUT'
	request = urllib2.urlopen(request)
	return request.read()